docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/hemoglobin/index.rst

```diff
@@ -6,4 +6,295 @@ Hemoglobin Model
+
+.. todo::
+
+	Integrate python code from above notebook into this document.
+
+.. code-block:: R
+
+  # define constants
+  XMAX = 220
+  EULERS_CONSTANT = 0.57721566490153286060651209008240243104215933593992
+  w = c(0.4,0.6)
+
+  # import standard R functions for the gamma distributions (pgamma and rgamma)
+  pacman::p_load(data.table,actuar)
+
+  # function to calculate gamma distribution parameters from mean and variance
+  gamma_mv2p = function(mn, vr){
+    list(shape = mn^2/vr,rate = mn/vr)}
+
+  # function to calculate mirror gumbel distribution parameters from mean and variance 
+  mgumbel_mv2p = function(mn, vr){
+    list(
+      alpha = XMAX - mn - EULERS_CONSTANT*sqrt(vr)*sqrt(6)/pi,
+      scale = sqrt(vr)*sqrt(6)/pi)}
+  # function to randomly sample n times from mirror gumbel distribution
+  rmgumbel = function(n, alpha, scale){
+    mn = alpha + scale*EULERS_CONSTANT
+    rgumbel(n, alpha+XMAX-(2*mn), scale)}
+
+  # function to sample from ensemble hemoglobin distribution using functions defined above
+    # n = number of samples to take
+    # mn = mean hemoglobin concentration
+    # vr = hemoglobin distribution variance (standard deviation squared)
+    # w = list of ensemble distribution weights c(gamma_weight, mirror_gumbel_weight)
+  sample_from_hemoglobin_distribution <- function(n, mn, vr, w){
+
+    ##parameters
+    params_gamma = gamma_mv2p(mn, vr)
+    params_mgumbel = mgumbel_mv2p(mn, vr)
+
+    ##weighting
+    sample_list = c(
+      rgamma(round(n * w[1], digits=0), params_gamma$shape, params_gamma$rate), 
+      rmgumbel(round(n * w[2], digits=0), params_mgumbel$alpha, params_mgumbel$scale)
+    )
+    sample_list
+  }
+
+  # function to calculate area under curve below threshold q for mirror gumbel distribution
+  pmgumbel = function(q, alpha, scale, lower.tail){ 
+    #NOTE: with mirroring, take the other tail
+    pgumbel(XMAX-q, alpha, scale, lower.tail=ifelse(lower.tail,FALSE,TRUE))}
+
+  # function to calculate area under curve of hemoglobin ensemble distribution using the functions defined above
+    # q = hemoglobin threshold
+    # mn = mean hemoglobin concentration
+    # vr = hemoglobin variance (standard deviation squared)
+    # w = list of ensemble distribution weights c(gamma_weight, mirror_gumbel_weight)
+  ens_mv2prev <- function(q, mn, vr, w){
+    x = q
+
+    ##parameters
+    params_gamma = gamma_mv2p(mn, vr)
+    params_mgumbel = mgumbel_mv2p(mn, vr)
+
+    ##weighting
+    prev = sum(
+      w[1] * pgamma(x, params_gamma$shape, params_gamma$rate), 
+      w[2] * pmgumbel(x, params_mgumbel$alpha, params_mgumbel$scale, lower.tail=T))
+    prev}
+
+.. note::
+
+	While not explicitly enforced by the code above, all hemoglobin values should be non-zero positive numbers. The probability of sampling a negative value is small, but if it occurs, the value should be resampled until it falls within the bounds of 0 and 220 or clipped to a value of 1.
```

Comment:
It's a bit confusing that this still mentions the upper bound 220 when now we're just talking about negative values. Maybe instead we could say "the value should be resampled until it is positive, or clipped to a value of 1."